data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/SearchAccessorStrategy.java

```diff
@@ -13,14 +59,242 @@
+    private final OpenSearchSourceConfiguration openSearchSourceConfiguration;
+
+    public static SearchAccessorStrategy create(final OpenSearchSourceConfiguration openSearchSourceConfiguration,
+                                                final AwsCredentialsSupplier awsCredentialsSupplier) {
+        return new SearchAccessorStrategy(openSearchSourceConfiguration, awsCredentialsSupplier);
+    }
+
+    private SearchAccessorStrategy(final OpenSearchSourceConfiguration openSearchSourceConfiguration,
+                                  final AwsCredentialsSupplier awsCredentialsSupplier) {
+        this.awsCredentialsSupplier = awsCredentialsSupplier;
+        this.openSearchSourceConfiguration = openSearchSourceConfiguration;
+    }
+
     /**
      * Provides a {@link SearchAccessor} that is based on the {@link OpenSearchSourceConfiguration}
-     * @param openSearchSourceConfiguration the plugins configuration
      * @return a {@link SearchAccessor}
      * @since 2.4
      */
-    public SearchAccessor getSearchAccessor(final OpenSearchSourceConfiguration openSearchSourceConfiguration) {
-        //todo: implement
-        return null;
+    public SearchAccessor getSearchAccessor() {
+        final RestClient restClient = createOpenSearchRestClient();
+        final OpenSearchTransport transport = createOpenSearchTransport(restClient);
+        final OpenSearchClient openSearchClient = new OpenSearchClient(transport);
+
+        InfoResponse infoResponse;
+        try {
+            infoResponse = openSearchClient.info();
+        } catch (final IOException | OpenSearchException e) {
+            throw new RuntimeException("There was an error looking up the OpenSearch cluster info: ", e);
+        }
+
+        final String distribution = infoResponse.version().distribution();
+        final String versionNumber = infoResponse.version().number();
+
+        if (!distribution.equals(OPENSEARCH_DISTRIBUTION)) {
+            throw new IllegalArgumentException(String.format("Only opensearch distributions are supported at this time. The cluster distribution being used is '%s'", distribution));
+        }
+
+        SearchContextType searchContextType;
+
+        if (versionSupportsPointInTimeForOpenSearch(versionNumber)) {
+            LOG.info("OpenSearch version {} detected. Point in time APIs will be used to search documents", versionNumber);
+            searchContextType = SearchContextType.POINT_IN_TIME;
+        } else {
+            LOG.info("OpenSearch version {} detected. Scroll contexts will be used to search documents. " +
+                    "Upgrade your cluster to at least version {} to use Point in Time APIs instead of scroll.", versionNumber, OPENSEARCH_POINT_IN_TIME_SUPPORT_VERSION_CUTOFF);
+            searchContextType = SearchContextType.SCROLL;
+        }
+
+        return new OpenSearchAccessor(openSearchClient, searchContextType);
+    }
+
+    private RestClient createOpenSearchRestClient() {
```

Comment:
The majority of these private methods were taken/based off of the opensearch sink (https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/ConnectionConfiguration.java)